### ISSUE TITLE
Input not file while creating App via Phonegap build API

### ISSUE THREAD

**Issue by malaiselvan** [mradionov/phonegap-build-api#3]
I use your API for creating and managing phonegap apps.  Till September 2015 your API was fine on creating apps.  From mid of September 2015 onwards it started showing '{"error":"Unable to create app: input is not a file"}'

I tried contact to phonegap community and stackoverflow
http://stackoverflow.com/questions/33881971/input-not-file-while-creating-app-via-phonegap-build-api
https://c.getsatisfaction.com/nitobi/topics/unable-to-create-app-input-is-not-a-file-df668pmjm93np

I don't think they are pointing me to the correct direction.  Is there any change required in this script?  Need your assistance.  Thanks.

**Comment by mradionov** [comment]
@malaiselvan , I am no longer using my lib to work with Phonegap Build API (I am away from PHP for a year or so), so I am not able to see if this library still works correctly. Thanks for your report, I'll definetely look into it.

**Comment by malaiselvan** [comment]
@mradionov Thanks for your swift response.  Much appreciated.  Thanks.

**Comment by mradionov** [comment]
@malaiselvan , I've stumbled upon a problem with CURL, seems that it no longer supports sending files in a format of `@path/to/file`, I get a PHP warning

```
Deprecated: curl_setopt(): The usage of the @filename API for file uploading is deprecated. Please use the CURLFile class instead in /var/www/github/phonegap-build-api/PhonegapBuildApi.php on line 828
```

Have you ever received this kind of warning? I was able to solve it temporary by modifying a line 814 in the library from:

```
$files[$key] = $value;
```

to
 
```
$files[$key] = new CurlFile(ltrim($value, '@'));
```

May I ask you to modify this line right in the library source in the project and tell me if it fixes the issue?
App is uploading for me from zip archive. This is the entire code I use (username and password changed):
```
<?php

include('PhonegapBuildApi.php');

$username = 'email@example.com';
$password = 'password';

$api = new PhonegapBuildApi($username, $password);
$response = $api->createApplicationFromFile('www.zip');

var_dump($response); // shows correct response

```

**Comment by mradionov** [comment]
@malaiselvan , nevermind, it's because my PHP is 5.5 and this `CurlFile` ss a feature of PHP 5.5. 
Actually, I am able to upload my zip archive to Phonegap Build, even without changing anything. I just get a deprecation warning.

**Comment by mradionov** [comment]
@malaiselvan , from StackOverflow issue you've mentioned - do you try to upload an archive from a link?

```
$data = array(
    "file" => "@http://mywebsite.com/tmp/www.tar.gz",
    //"file" => "@actual path to/www.tar.gz",
    "data" => '{"title":"MyTestApp","create_method":"file"}'
  );
```

Did it always work for you?

**Comment by malaiselvan** [comment]
Yes, it always works for me.

I changed the code 

    $files[$key] = new CurlFile(ltrim($value, '@'));

and the file got uploaded for me now.  I will do few more tests and will update on you.

**Comment by malaiselvan** [comment]
Hi, I can confirm that the change to CurlFile works great.  You really solved my problem. Million of Thanks.

**Comment by malaiselvan** [comment]
Closing this issue.

**Comment by mradionov** [comment]
Sounds great. I'll re-open it because I want to land it as a fix in the repository. BTW, do you use PHP 5.5 too? Is this issue somehow related to a PHP upgrade on your servers?

**Comment by malaiselvan** [comment]
I checked and confirm that I use 5.6.15.  I use a shared hosting from hostmonster but not sure whether the server guys upgraded my PHP.

I tried few uploads now with the new code and it works great for me.

**Comment by mradionov** [comment]
I guess it was a deprecation in 5.5, and was removed in 5.6. Now it makes sense.
For more information:
* http://stackoverflow.com/questions/25934128/curl-file-uploads-not-working-anymore-after-upgrade-from-php-5-5-to-5-6
* http://stackoverflow.com/questions/20972513/php-curl-the-usage-of-the-filename-api-for-file-uploading-is-deprecated

**Comment by mradionov** [comment]
I've released a fix in version **0.1.1**, but I guess you can still use **dev-master** in composer.
As an extra I've added documentation. @malaiselvan , thanks again for reporting the issue!
